src/lib.rs

```diff
@@ -400,6 +400,307 @@ pub extern "C" fn DATE_AND_TIME_TO_TIME_OF_DAY(input: i64) -> i64 {
     new_date_time.timestamp_millis()
 }
 
+/// .
+/// This operator returns the value of adding up two TIME operands.
+/// Panic on overflow
+///
+#[allow(non_snake_case)]
+#[no_mangle]
+pub extern "C" fn ADD_TIME(in1: i64, in2: i64) -> i64 {
+    chrono::Duration::milliseconds(in1)
+        .checked_add(&chrono::Duration::milliseconds(in2))
+        .unwrap()
+        .num_milliseconds()
+}
+
+/// .
+/// This operator returns the value of adding up TOD and TIME.
+/// Panic on onverflow
+///
+#[allow(non_snake_case)]
+#[no_mangle]
+pub extern "C" fn ADD_TOD_TIME(in1: i64, in2: i64) -> i64 {
+    add_datetime_time(in1, in2)
+}
+
+/// .
+/// This operator returns the value of adding up DT and TIME.
+/// Panic on overflow
+///
+#[allow(non_snake_case)]
+#[no_mangle]
+pub extern "C" fn ADD_DT_TIME(in1: i64, in2: i64) -> i64 {
+    add_datetime_time(in1, in2)
+}
+
+#[allow(non_snake_case)]
+#[no_mangle]
+pub extern "C" fn add_datetime_time(in1: i64, in2: i64) -> i64 {
```

Comment:
If this is also called in ST I think it should be all capital, otherwise this is a snake_case name and it does not need the #[allow]
It also would not need the extern C and no mangle